pynest/examples/glif_psc_neuron.py

```diff
@@ -0,0 +1,322 @@
+                               "voltage_reset_fraction": 0.20,
+                               "spike_dependent_threshold": mechs[0],  # True
+                               "after_spike_currents": mechs[1],  # True
+                               "adapting_threshold": mechs[2]})  # False
+
+
+def create_lif_r_asc_a_psc(mechs, syn_tau):
+    """Creates a nest glif_psc object for GLIF Level 5 (LIF-R-ASC-A) neuron"""
+    return nest.Create("glif_psc",
+                       params={"C_m": 58.72,
+                               "E_L": -78.85,
+                               "V_m": -78.85,
+                               "V_th": -51.68,
+                               "asc_amps": [-9.18, -198.94],
+                               "asc_decay": [0.003, 0.1],
+                               "asc_init": [0.0, 0.0],
+                               "asc_r": [1.0, 1.0],
+                               "g": 9.43,
+                               "t_ref": 3.75,
+                               "tau_syn": syn_tau,
+                               "th_spike_add": 0.37,
+                               "th_spike_decay": 0.009,
+                               "th_voltage_decay": 0.09,
+                               "th_voltage_index": 0.005,
+                               "voltage_reset_add": 18.56,
+                               "voltage_reset_fraction": 0.20,
+                               "spike_dependent_threshold": mechs[0],  # True
+                               "after_spike_currents": mechs[1],  # True
+                               "adapting_threshold": mechs[2]})  # True
+
+##############################################################################
+# We pre-define model mechanism parameters for each level of GLIF models
+# in the order of[spike_dependent_threshold, after_spike_currents,
+# adapting_threshold].We also pre - define the synapse time constant array,
+# [2.0, 1.0] ms for two desgired synaptic ports of the neuron.
+
+model_mech_dict = {"lif": [False, False, False],
+                   "lif_r": [True, False, False],
+                   "lif_asc": [False, True, False],
+                   "lif_r_asc": [True, True, False],
+                   "lif_r_asc_a": [True, True, True]}
+
+syn_tau = [2.0, 1.0]
+
+###############################################################################
+# We pick one of the five GLIF model levels to be tested from
+# ("lif", "lif_r", "lif_asc", "lif_r_asc", "lif_r_asc_a").
+# Test "lif" model if name other than the five above names is set.
+# For the picked GLIF model, model mechanisms are retrieved to create
+# ``glif_psc`` node.The node is created by calling the above defined function
+# for the picked model.We store the returned handle in variable for later
+# reference.
+
+glif_model = "lif"  # "lif", "lif_r", "lif_asc", "lif_r_asc", or "lif_r_asc_a"
+
+if glif_model not in model_mech_dict.keys():
+    glif_model = 'lif'
+
+mechs = model_mech_dict[glif_model]
```

Comment:
I think the example would be more instructive if you created one neuron of each variant and the showed how they react to the same input, i.e., something like
```python
n_lif = nest.Create('glif_psc', 'spike_dependent_threshold': False, 'after_spike_currents': False, 'adapting_threshold': False)
n_lif_r = nest.Create('glif_psc', 'spike_dependent_threshold': True, 'after_spike_currents': False, 'adapting_threshold': False)
...
neurons = n_lif + n_lif_r + ...
```